google/apistore.go

```diff
@@ -1,22 +1,26 @@
-package cloudstorage
+package google
 
-import "google.golang.org/api/storage/v1"
+import (
+	"github.com/lytics/cloudstorage"
+	"google.golang.org/api/storage/v1"
+)
 
+// ApiStore a google api store
 type ApiStore struct {
```

Comment:
Regarding the two methods: `BucketExists` and `CreateBucket`:  This could eventually become an interface of type `StoreManager` or something that has methods for configuring the store. 

Regarding the two methods: `AddOwner` and `AddReader`: I think we should follow the pattern you used for `Move` and make these an optional interface .